src/components/Popup/Popup.ts

```diff
@@ -0,0 +1,127 @@
+import { nostojs } from "@nosto/nosto-js"
+import { customElement } from "../decorators"
+import { NostoElement } from "../Element"
+import { popupStyles } from "./styles"
+
+/**
+ * A custom element that displays popup content with dialog and ribbon slots.
+ * Supports conditional activation based on Nosto segments and persistent closure state.
+ *
+ * @property {string} [name] - Optional name used for analytics and localStorage persistence. When provided, the popup's closed state will be remembered.
+ * @property {string} [segment] - Optional Nosto segment that acts as a precondition for activation. Only users in this segment will see the popup.
+ *
+ * @example
+ * Basic popup with dialog and ribbon content:
+ * ```html
+ * <nosto-popup name="promo-popup" segment="5b71f1500000000000000006">
+ *   <div slot="default">
+ *     <h2>Special Offer!</h2>
+ *     <p>Get 20% off your order today</p>
+ *     <button n-close>Close</button>
+ *   </div>
+ *   <div slot="ribbon">
+ *     <span>Limited time!</span>
+ *   </div>
+ * </nosto-popup>
+ * ```
+ */
+@customElement("nosto-popup")
+export class Popup extends NostoElement {
+  /** @private */
+  static attributes = {
+    name: String,
+    segment: String
+  }
+
+  name?: string
+  segment?: string
+
+  constructor() {
+    super()
+    this.attachShadow({ mode: "open" })
+  }
+
+  async connectedCallback() {
+    if (!(await isPopupShown(this))) {
+      this.style.display = "none"
+      return
+    }
+
+    renderShadowContent(this)
+    this.addEventListener("click", this.handleClick)
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener("click", this.handleClick)
+  }
+
+  private handleClick = (event: Event) => {
+    const target = event.target as HTMLElement
+    const closeElement = target?.closest("[n-close]")
+    if (closeElement) {
+      event.preventDefault()
+      event.stopPropagation()
+      closePopup(this)
+    }
+  }
+}
+
+function renderShadowContent(element: Popup) {
```

Comment:
pass mode: "open" | "ribbon"
in open mode, the dialog should be shown and ribbon should be hidden
in ribbon mode, the dialog should be closed and ribbon should be shown